consensus/raft/data_helper_test.go

```diff
@@ -0,0 +1,34 @@
+package raft
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+func TestDataBackupHelper(t *testing.T) {
+	cleanup := func() {
+		os.RemoveAll("data_helper_testing")
+		for i := 0; i < 2*RaftDataBackupKeep; i++ {
+			os.RemoveAll(fmt.Sprintf("data_helper_testing.old.%d", i))
+		}
+	}
+	cleanup()
+	defer cleanup()
+
+	os.MkdirAll("data_helper_testing", 0700)
+	helper := newDataBackupHelper("data_helper_testing")
+	for i := 0; i < 2*RaftDataBackupKeep; i++ {
+		err := helper.makeBackup()
+		if err != nil {
+			t.Fatal(err)
+		}
+		backups := helper.listBackups()
+		if i < RaftDataBackupKeep && len(backups) != i+1 {
+			t.Fatal("not saving enough backups")
```

Comment:
If something really weird happened with the helper and too many backups were saved before iteration `RaftDataBackupKeep` or alternatively too few backups were saved after iteration `RaftDataBackupKeep`, then the error messages will be misleading.  As the check stands, a better message would be "incorrect number of backups saved"

Reply:
ok fixed